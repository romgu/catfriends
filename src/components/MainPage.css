h1 {
	font-display: swap;
	font-family: 'Playfair Display', serif;
}

.marginTop {
	margin: 3vh 0px 0px 0px;
}