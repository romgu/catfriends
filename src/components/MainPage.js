import React, { Component } from 'react';

import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import { Scrollbars } from 'react-custom-scrollbars';
import Header from '../components/Header'

import './MainPage.css';

class MainPage extends Component {
	componentDidMount() {
		this.props.onRequestCats();
	}

	filteredCats = () => {
		return this.props.cats.filter(cat => {
			return cat.name.toLowerCase().includes(this.props.searchField.toLowerCase());
		});
	}

	render () {
		const { onSearchChange, cats, isPending } = this.props;
		return isPending ? 
		<h1>Loading</h1> :
		(
			<div className="tc">
				<Header />
				<SearchBox searchChange={onSearchChange} />
				<Scrollbars style={{ width:'100%', height:'80vh' }}>
					<CardList cats={this.filteredCats()} />
				</Scrollbars>
			</div>
		);	
	}
}


export default MainPage;