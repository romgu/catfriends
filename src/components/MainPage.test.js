import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';



let wrapper;
beforeEach(() => {
	const mockProps = {
		onRequestCats: jest.fn(),
		cats: [],
		searchField: '',
		isPending: false,
	}
	wrapper = shallow(<MainPage {...mockProps}/>)
})

it('renders MainPage without crashing', () => {
	expect(wrapper).toMatchSnapshot();
})

it('filter cats correctly', () => {
	expect(wrapper.instance().filteredCats([])).toEqual([])
})

it('filter cats correctly', () => {
	const mockProps2 = {
		onRequestCats: jest.fn(),
		cats: [{
			id: 3,
			name: 'john',
			email: 'john@gmail.com'
		}],
		searchField: 'john',
		isPending: false,
	}
	const wrapper2 = shallow(<MainPage {...mockProps2}/>)
	expect(wrapper2.instance().filteredCats()).toEqual([{
			id: 3,
			name: 'john',
			email: 'john@gmail.com'
		}])
})