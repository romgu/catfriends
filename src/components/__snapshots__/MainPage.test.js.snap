// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    cats={Array []}
    isPending={false}
    onRequestCats={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    searchField=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <SearchBox />,
        <Scrollbars
          autoHeight={false}
          autoHeightMax={200}
          autoHeightMin={0}
          autoHide={false}
          autoHideDuration={200}
          autoHideTimeout={1000}
          hideTracksWhenNotNeeded={false}
          renderThumbHorizontal={[Function]}
          renderThumbVertical={[Function]}
          renderTrackHorizontal={[Function]}
          renderTrackVertical={[Function]}
          renderView={[Function]}
          style={
            Object {
              "height": "80vh",
              "width": "100%",
            }
          }
          tagName="div"
          thumbMinSize={30}
          universal={false}
        >
          <CardList
            cats={Array []}
          />
        </Scrollbars>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchChange": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoHeight": false,
          "autoHeightMax": 200,
          "autoHeightMin": 0,
          "autoHide": false,
          "autoHideDuration": 200,
          "autoHideTimeout": 1000,
          "children": <CardList
            cats={Array []}
          />,
          "hideTracksWhenNotNeeded": false,
          "renderThumbHorizontal": [Function],
          "renderThumbVertical": [Function],
          "renderTrackHorizontal": [Function],
          "renderTrackVertical": [Function],
          "renderView": [Function],
          "style": Object {
            "height": "80vh",
            "width": "100%",
          },
          "tagName": "div",
          "thumbMinSize": 30,
          "universal": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cats": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <SearchBox />,
          <Scrollbars
            autoHeight={false}
            autoHeightMax={200}
            autoHeightMin={0}
            autoHide={false}
            autoHideDuration={200}
            autoHideTimeout={1000}
            hideTracksWhenNotNeeded={false}
            renderThumbHorizontal={[Function]}
            renderThumbVertical={[Function]}
            renderTrackHorizontal={[Function]}
            renderTrackVertical={[Function]}
            renderView={[Function]}
            style={
              Object {
                "height": "80vh",
                "width": "100%",
              }
            }
            tagName="div"
            thumbMinSize={30}
            universal={false}
          >
            <CardList
              cats={Array []}
            />
          </Scrollbars>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchChange": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoHeight": false,
            "autoHeightMax": 200,
            "autoHeightMin": 0,
            "autoHide": false,
            "autoHideDuration": 200,
            "autoHideTimeout": 1000,
            "children": <CardList
              cats={Array []}
            />,
            "hideTracksWhenNotNeeded": false,
            "renderThumbHorizontal": [Function],
            "renderThumbVertical": [Function],
            "renderTrackHorizontal": [Function],
            "renderTrackVertical": [Function],
            "renderView": [Function],
            "style": Object {
              "height": "80vh",
              "width": "100%",
            },
            "tagName": "div",
            "thumbMinSize": 30,
            "universal": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "cats": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
